Warning in rheology/composite_visco_plastic.cc

```
/ssd/aspect-git5/source/material_model/rheology/drucker_prager.cc:80:45: warning: ‘drucker_prager_parameters.aspect::MaterialModel::Rheology::DruckerPragerParameters::cohesion’ may be used uninitialized [-Wmaybe-uninitialized]
/ssd/aspect-git5/source/material_model/rheology/composite_visco_plastic.cc: In member function ‘double aspect::MaterialModel::Rheology::CompositeViscoPlastic<dim>::compute_composition_viscosity(double, double, unsigned int, const dealii::SymmetricTensor<2, dim>&, std::vector<double>&, const std::vector<double>&, const std::vector<unsigned int>&) const [with int dim = 3]’:
/ssd/aspect-git5/source/material_model/rheology/composite_visco_plastic.cc:139:43: note: ‘drucker_prager_parameters.aspect::MaterialModel::Rheology::DruckerPragerParameters::cohesion’ was declared here
```

--- a/include/aspect/material_model/rheology/diffusion_creep.h
+++ b/include/aspect/material_model/rheology/diffusion_creep.h
@@ -48,6 +48,11 @@ namespace aspect
         double activation_volume;
         double stress_exponent;
         double grain_size_exponent;
+
+        /**
+         * Constructor. Initializes all values to NaN.
+         */
+        DiffusionCreepParameters();
       };
 
       template <int dim>

--- a/include/aspect/material_model/rheology/dislocation_creep.h
+++ b/include/aspect/material_model/rheology/dislocation_creep.h
@@ -45,6 +45,11 @@ namespace aspect
         double activation_energy;
         double activation_volume;
         double stress_exponent;
+
+        /**
+         * Constructor. Initializes all values to NaN.
+         */
+        DislocationCreepParameters();
       };
 
       template <int dim>

--- a/include/aspect/material_model/rheology/drucker_prager.h
+++ b/include/aspect/material_model/rheology/drucker_prager.h
@@ -54,6 +54,11 @@ namespace aspect
          * Limit maximum yield stress from drucker prager yield criterion.
          */
         double max_yield_stress;
+
+        /**
+         * Constructor. Initializes all values to NaN.
+         */
+        DruckerPragerParameters();
       };
 
       template <int dim>

--- a/include/aspect/material_model/rheology/peierls_creep.h
+++ b/include/aspect/material_model/rheology/peierls_creep.h
@@ -51,6 +51,11 @@ namespace aspect
         double glide_parameter_q;
         double fitting_parameter;
         double stress_cutoff;
+
+        /**
+         * Constructor. Initializes all values to NaN.
+         */
+        PeierlsCreepParameters();
       };
 
       /**

--- a/source/material_model/rheology/composite_visco_plastic.cc
+++ b/source/material_model/rheology/composite_visco_plastic.cc
@@ -137,6 +137,7 @@ namespace aspect
         Rheology::DislocationCreepParameters dislocation_creep_parameters;
         Rheology::PeierlsCreepParameters peierls_creep_parameters;
         Rheology::DruckerPragerParameters drucker_prager_parameters;
+
         double eta_diff = maximum_viscosity;
         double eta_disl = maximum_viscosity;
         double eta_prls = maximum_viscosity;

--- a/source/material_model/rheology/diffusion_creep.cc
+++ b/source/material_model/rheology/diffusion_creep.cc
@@ -33,6 +33,16 @@ namespace aspect
   {
     namespace Rheology
     {
+      DiffusionCreepParameters::DiffusionCreepParameters()
+        : prefactor (numbers::signaling_nan<double>()),
+          activation_energy (numbers::signaling_nan<double>()),
+          activation_volume (numbers::signaling_nan<double>()),
+          stress_exponent (numbers::signaling_nan<double>()),
+          grain_size_exponent (numbers::signaling_nan<double>())
+      {}
+
+
+
       template <int dim>
       DiffusionCreep<dim>::DiffusionCreep ()
         = default;

--- a/source/material_model/rheology/dislocation_creep.cc
+++ b/source/material_model/rheology/dislocation_creep.cc
@@ -32,6 +32,15 @@ namespace aspect
   {
     namespace Rheology
     {
+      DislocationCreepParameters::DislocationCreepParameters()
+        :  prefactor (numbers::signaling_nan<double>()),
+           activation_energy (numbers::signaling_nan<double>()),
+           activation_volume (numbers::signaling_nan<double>()),
+           stress_exponent (numbers::signaling_nan<double>())
+      {}
+
+
+
       template <int dim>
       DislocationCreep<dim>::DislocationCreep ()
         = default;

--- a/source/material_model/rheology/drucker_prager.cc
+++ b/source/material_model/rheology/drucker_prager.cc
@@ -32,9 +32,19 @@ namespace aspect
   {
     namespace Rheology
     {
+      DruckerPragerParameters::DruckerPragerParameters()
+        : angle_internal_friction (numbers::signaling_nan<double>()),
+          cohesion  (numbers::signaling_nan<double>()),
+          max_yield_stress (numbers::signaling_nan<double>())
+      {}
+
+
+
       template <int dim>
       DruckerPrager<dim>::DruckerPrager ()
         = default;
+
+
 
       template <int dim>
       const DruckerPragerParameters

--- a/source/material_model/rheology/peierls_creep.cc
+++ b/source/material_model/rheology/peierls_creep.cc
@@ -32,6 +32,20 @@ namespace aspect
   {
     namespace Rheology
     {
+      PeierlsCreepParameters::PeierlsCreepParameters()
+        : prefactor (numbers::signaling_nan<double>()),
+          stress_exponent (numbers::signaling_nan<double>()),
+          activation_energy (numbers::signaling_nan<double>()),
+          activation_volume (numbers::signaling_nan<double>()),
+          peierls_stress (numbers::signaling_nan<double>()),
+          glide_parameter_p (numbers::signaling_nan<double>()),
+          glide_parameter_q (numbers::signaling_nan<double>()),
+          fitting_parameter (numbers::signaling_nan<double>()),
+          stress_cutoff (numbers::signaling_nan<double>())
+      {}
+
+
+
       template <int dim>
       PeierlsCreep<dim>::PeierlsCreep ()
         = default;

